src/signature.rs

```diff
@@ -9,33 +9,23 @@
 
 //! An ed25519 signature.
 
-use core::fmt::Debug;
+#![allow(non_snake_case)]
+#![allow(absolute_paths_not_starting_with_crate)]
 
 use curve25519_dalek::edwards::CompressedEdwardsY;
 use curve25519_dalek::scalar::Scalar;
 
-#[cfg(feature = "serde")]
-use serde::de::Error as SerdeError;
-#[cfg(feature = "serde")]
-use serde::de::Visitor;
-#[cfg(feature = "serde")]
-use serde::{Deserialize, Serialize};
-#[cfg(feature = "serde")]
-use serde::{Deserializer, Serializer};
-
 use crate::constants::*;
-use crate::errors::*;
 
-/// An ed25519 signature.
-///
-/// # Note
-///
-/// These signatures, unlike the ed25519 signature reference implementation, are
-/// "detached"—that is, they do **not** include a copy of the message which has
-/// been signed.
-#[allow(non_snake_case)]
-#[derive(Copy, Eq, PartialEq)]
-pub struct Signature {
+// Re-export the `ed25519` crate's Signature type
+pub use ed25519_crate::{Signature, signature::Signature as SignatureTrait};
+
+/// Extension trait for the `Signature` type providing access to the
+/// `R` and `s` components of the signature.
```

Comment:
Since the `ed25519` crate provides the `ed25519::Signature` type (which internally is a 64-byte "octet string"), I opted to define an extension trait to provide the previous behavior. Let me know what you think of this approach. Alternatively I think this code could be factored to the relevant call sites.